version: "3"
services:
    lora_d2d:
        build:
            context: .
            dockerfile: Dockerfile

        volumes:
          - ./results:/results
          - ./lora:/root/ns-allinone-3.31/ns-3.31/src/lora/
          - ./lorawan:/root/ns-allinone-3.31/ns-3.31/src/lorawan/
          - ./experiments:/root/ns-allinone-3.35/ns-3.35/scratch/lora_d2d/
          - ./experiments:/root/ns-allinone-3.35/ns-3.35/experiments
        environment:
          - RUN
          - LOG_PATH
    jupyter:
        image: jupyter/datascience-notebook:latest
        container_name: lora_d2d_jupyter
        ports:
            - "8888:8888"
        volumes:
            - ./eval/:/home/docker_user/work/eval
            - ./results/:/home/docker_user/work/results
        environment:
            JUPYTER_ENABLE_LAB: "yes"
            JUPYTER_TOKEN: "lora_d2d"
            NB_USER: docker_user
            NB_UID: 2016
            NB_GID: 2001
            CHOWN_HOME: 'yes'
            CHOWN_HOME_OPTS: -R
